device.go

```diff
@@ -172,9 +125,12 @@ func (device *VFIODevice) attach(h hypervisor) error {
 
 		device.BDF = deviceBDF
 
-		if err := h.addDevice(device, vfioDev); err != nil {
+		if err := h.addDevice(*device, vfioDev); err != nil {
```

Comment:
This is a separate fix, please submit it as a separate patch.

Reply:
Done